src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -67,6 +76,18 @@ class TezosPlatformDiscoveryOperations(
     result.map(_.toMap)
   }
 
+  private def preCacheAttributeValues: DBIO[AttributeValuesCache] = {
+    DBIO.sequence {
+      cacheOverrides.getAttributesToCache.map {
+        case (table, column) =>
+          TezosDatabaseOperations.selectDistinct(table, column).map { values =>
+            val radixTree = RadixTree(values.map(x => x.toLowerCase -> x): _*)
```

Comment:
Why the RadixTree over other indexing structures? Is it because block_ids are the most problematic field, and the relative character distance between block_ids is small?

Reply:
I used radix trees because they are very efficient when searching for multiple results with given prefix - that was crucial because we want to use it for filtering in autocomplete feature. At first I wanted to use regular prefix tree(also known as Trie), but I've found library which already implemented RadixTree.

Reply:
What other indexing structures did you have in mind?

Reply:
That makes sense, I just wasn't 100% understanding the spec for what we needed, thanks for the clarification.